src/cascade/dismod/db/wrapper.py

```diff
@@ -74,6 +76,35 @@ def _validate_data(table_definition, data):
         raise DismodFileError(f"extra columns in data for table '{table_definition.name}': {extra_columns}")
 
 
+def _ordered_by_foreign_key_dependency(schema, tables_to_write):
+    """
+    Iterator that walks through tables in the correct order for foreign
+    keys to be written before those that depend upon them.
+
+    Args:
+        schema (sqlalchemy.metadata): Metadata about the tables.
+        tables_to_write (iterable[str]): An iterable of table names as string.
+
+    Returns:
+        Iterates through the tables in an order that is safe for writing.
+    """
+    dependency_graph = DiGraph()
+
+    if len(set(tables_to_write) - set(schema.tables.keys())) > 0:
+        raise ValueError("Asking to write tables not in schema")
+
+    for scan_name, scan_table in schema.tables.items():
+        dependency_graph.add_node(scan_name)
+        for foreign_key in scan_table.foreign_keys:
+            target_name = foreign_key.target_fullname.split(".")[0]
+            dependency_graph.add_edge(target_name, scan_name)
+
+    # Use the full lexicographical sort because it makes testing deterministic.
```

Comment:
As opposed to what?

Reply:
There is a purely topological sort, but then the order of tables won't be deterministic. The lexicographical (within toplogical constraints) is deterministic.

Reply:
I see. That makes sense.